core/src/rpc_subscriptions.rs

```diff
@@ -11,17 +11,47 @@ use solana_sdk::{
 };
 use solana_vote_program::vote_state::MAX_LOCKOUT_HISTORY;
 use std::collections::HashMap;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::mpsc::{Receiver, SyncSender, TryRecvError, TrySendError};
 use std::sync::{Arc, RwLock};
+use std::thread::{Builder, JoinHandle};
+use std::time::Duration;
+
+const NOTIFICATION_QUEUE_SIZE: usize = 1_000;
+const RECV_DELAY_MILLIS: u64 = 100;
 
 pub type Confirmations = usize;
 
-#[derive(Serialize, Clone)]
+#[derive(Serialize, Clone, Copy, Debug)]
 pub struct SlotInfo {
     pub slot: Slot,
     pub parent: Slot,
     pub root: Slot,
 }
 
+// FEEDBACK: proper use of unsafe?
```

Comment:
Why do we need these unsafes?

Reply:
* it's sending the `NotificationEntry` instances through the channel, which I believe requires Send/Sync?
* or, is the answer something simple, should I just be sending `.clone()`s through?

Reply:
How about wrapping it in an Arc?